devito/passes/clusters/cse.py

```diff
@@ -98,24 +98,26 @@ def _cse(maybe_exprs, make, min_cost=1, mode='default'):
 
         # Create temporaries
         hit = max(targets.values())
-        temps = [Eq(make(), k) for k, v in targets.items() if v == hit]
+        chosen = [(k, make()) for k, v in targets.items() if v == hit]
 
         # Apply replacements
         # The extracted temporaries are inserted before the first expression
         # that contains it
+        scheduled = []
         updated = []
         for e in processed:
             pe = e
-            for t in temps:
-                pe, changed = _uxreplace(pe, {t.rhs: t.lhs})
-                if changed and t not in updated:
-                    updated.append(t)
+            for k, v in chosen:
+                pe, changed = _uxreplace(pe, {k: v})
+                if changed and v not in scheduled:
+                    updated.append(pe.func(v, k, operation=None))
+                    scheduled.append(v)
             updated.append(pe)
         processed = updated
```

Comment:
Why not just use `updated` henceforth, rather than using `processed`?

Reply:
wdym ? we keep on looping and `processed` needs to be updated? what change are you proposing ?

Reply:
Ignore me then

Reply:
Missed an indent